src/ui/pages/SendTransaction/NetworkFee/NetworkFeeDialog.tsx

```diff
@@ -112,15 +113,22 @@ function setFormValue(form: HTMLFormElement, name: string, value: unknown) {
   }
 }
 
-function setPatternValidity(event: React.ChangeEvent<HTMLInputElement>) {
-  if (event.currentTarget.validity.patternMismatch) {
-    event.currentTarget.setCustomValidity(
-      'Gas Price value must be a positive number'
-    );
-  } else {
-    event.currentTarget.setCustomValidity('');
-  }
-}
+const setPatternValidity =
+  (message: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.currentTarget.validity.patternMismatch) {
+      event.currentTarget.setCustomValidity(message);
+    } else {
+      event.currentTarget.setCustomValidity('');
+    }
+  };
+
+const setGasPriceValidationMessage = setPatternValidity(
+  'Gas Price value must be a positive number'
+);
+
+const setGasLimitValidationMessage = setPatternValidity(
+  'Gas Limit value must be a natural number'
```

Comment:
Not sure about the "Natural number" term. Is it indeed correct, but is it a well-known? :)

Reply:
The alternative is "integer" which also seems technical, so i'm not sure :)